sympy/core/numbers.py

```diff
@@ -3879,9 +3882,7 @@ def _eval_evalf(self, prec):
         return Float._new(self._as_mpf_val(prec), prec)
 
     def __eq__(self, other):
-        try:
-            other = _sympify(other)
-        except SympifyError:
+        if not isinstance(other, Basic):
             return NotImplemented
```

Comment:
Won't this mean that one gets a NotImplemented doing e.g `zoo == 1`? Or am I missing something?

Reply:
if a.__eq__(b) returns `NotImplemented`, then it will check b.__eq__(a), if both return `NotImplemented` it will fall back to identity comparison (I believe this is `a is b`, which will probably be `False`). Basically it allows an arbitrary other class to define a sensible `a == b` and `b == a` without having to modify both classes.

Reply:
I now changed it to use `Basic.__eq__` because someone might have user defined conversions (e.g. in `converter` or because `other` has `_sympy_` defined. Although that method also returns `NotImplemented` when it doesn't know the type of `other`.

Reply:
Ahh, all right! It is not `NotImplemtedError.